Changelog.python.md

```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## Python Icechunk Library 0.1.0a10
+
+### Features
+
+- Large chunks are fetched concurrently
```

Comment:
```suggestion
- Large chunks are fetched concurrently
- `IcechunkStore.list_dir` is now significantly faster
```

Reply:
Let's put it under the Performance section